chore(kno-9950): Clarify Slack integration capabilities (#1158)
knocklabs/docs: content/integrations/chat/slack/overview.mdx

This page covers how to use Knock to send notifications to Slack. Depending on your use case, there are a few different ways to approach this integration. This documentation serves as a starting point and will cover the basics of setting up a Slack integration with Knock regardless of your use case.

## What can I do with Knock's Slack integration?

> Knock's Slack integration enables you to send notifications to [Slack channels](https://docs.knock.app/integrations/chat/slack/sending-a-message-to-channels) and [direct messages](https://docs.knock.app/integrations/chat/slack/sending-a-direct-message) to users, or use [incoming webhooks for internal workspaces](https://docs.knock.app/integrations/chat/slack/sending-an-internal-message). You can create rich message templates using [markdown](#markdown-templates) or [Slack's block kit framework](#block-based-templates), and build self-serve integrations where customers connect their own Slack workspaces.

Reviewer: ```suggestion
Knock's Slack integration enables you to send notifications to [Slack channels](https://docs.knock.app/integrations/chat/slack/sending-a-message-to-channels) and [direct messages](https://docs.knock.app/integrations/chat/slack/sending-a-direct-message) to users via an authenticated Slack app, or via [webhooks for internal workspaces](https://docs.knock.app/integrations/chat/slack/sending-an-internal-message). You can create rich message templates using [markdown](#markdown-templates) or [Slack's block kit framework](#block-based-templates), and build integrations where customers connect your Slack app to their Slack workspaces.
```

Reasons for the suggestions:
- even though these are called "incoming webhooks" in Slack, thought it might be helpful to differentiate here that those webhooks would not be incoming to Knock
- self-serve sounded a bit like building the integration was going to be a really quick thing, feel free to adjust the wording  but I thought maybe this is a bit clearer?
